stacmodel/src/main/java/au/org/aodn/stac/model/ConceptModel.java

```diff
@@ -0,0 +1,32 @@
+package au.org.aodn.stac.model;
+
+import lombok.Builder;
+import lombok.Data;
+
+import java.util.Objects;
+
+
+@Data
+@Builder
+public class ConceptModel {
+    private String id;
+    private String url;
+
+    public ConceptModel(String id, String url) {
+        this.id = id;
+        this.url = url;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```

Comment:
use instanceof easier?

Reply:
I want to compare two different objects of the same class `ConceptModel` that be created/initialised differently rather than checking if the objects are of type `ConceptModel`. So `instanceof` may not be helpful: https://www.baeldung.com/java-comparing-objects#equals-instance

Reply:
well depends @utas-raymondng , if you extend the `ConceptModel` class with extra field such as `somethingElse`, but you don't change how the `equal` method is defined, it will still comparing against `id` and `url` ; I've seen most of the entities of the Animal Tracking project using this way to compare the objects, e.g https://github.com/aodn/animal-tracking/blob/a5fdc7741246554ec75503813fdbd9217279af71/animal-tracking-rest-api/src/main/java/au/org/aodn/animaltracking/restapi/model/DetectionEntity.java#L166-L184

```java
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ConceptModel that = (ConceptModel) o;
        return Objects.equals(id, that.id) &&
                Objects.equals(url, that.url);
    }
```